File: src/structs/loader.rs
#[derive(Debug, Clone, Copy, Default, PartialEq)]
pub enum Loader {
    Fabric,
    Forge,
    Quilt,
    #[default]
    Unknown,
}

impl ToString for Loader {
    fn to_string(&self) -> String {
        match self {
            Loader::Fabric => "fabric",
            Loader::Forge => "forge",
            Loader::Quilt => "quilt",
            Loader::Unknown => "unknown",
        }
        .to_string()
    }
}

File: src/lib/loader.ts
export enum Loader {
	Forge = 0,
	Fabric = 1,
	Quilt = 2,
	Neoforge = 3,
}

export function loaderToString(loader: Loader) {
	switch (loader) {
		case Loader.Forge:
			return 'forge';
		case Loader.Fabric:
			return 'fabric';
		case Loader.Quilt:
			return 'quilt';
		case Loader.Neoforge:
			return 'neoforge';
	}
}
